### ISSUE TITLE
Errors on run if ius-release package is already installed (CentOS 6)

### ISSUE THREAD

**Issue by davidjb** [saltstack/salt-bootstrap#1121]
### Description of Issue/Question

I'm trying to restore automated Salt installation after experiencing #1115 -- currently salt-bootstrap is broken for me on CentOS 6.

In trying to workaround #1115, I followed @rallytime's suggestions to try and get things working.  What I found is that with the `-y` option enabled, if the `ius-release` package is already installed (eg I had it already or else if you'd already run salt-bootstrap), salt-bootstrap errors out at the point of checking dependencies.  

A further workaround is to use the `-r` option to disable repository checking, but this means some foreknowledge about the target system is required to decide whether or not to use that option.  

### Setup
Run the following CentOS 6.7 x86_64 VM:
```
curl -o bootstrap-salt.sh -L https://bootstrap.saltstack.com/develop
sudo sh bootstrap-salt.sh -D -P -y -x python2.7 git develop
```

### Steps to Reproduce Issue
Run the above, wait while the setup takes place and note the error at the end:
```
INFO: Cloning Salt's git repository succeeded
INFO: Attempting to install a repo to help provide a separate python package
INFO: centos
INFO: Installing IUS repo
Loaded plugins: fastestmirror
Setting up Install Process
Examining /var/tmp/yum-root-WwmIVv/ius-release.rpm: ius-release-1.0-15.ius.centos6.noarch
/var/tmp/yum-root-WwmIVv/ius-release.rpm: does not update installed package.
Error: Nothing to do
ERROR: Failed to run install_centos_git_deps()!!!
DEBUG: Cleaning up the Salt Temporary Git Repository
DEBUG: Removing the logging pipe /tmp/bootstrap-salt.logpipe
```

Running the `yum install .../ius-release.rpm` command directly return `1` so this appears to be what's being detected as an error.

### Versions and Systems

`bootstrap-salt.sh -v`
```
develop -- Version 2017.05.24
```

Adding the `-r` option to install (so that install succeeds) allows me to run `salt --versions-report`:
```
Salt Version:
           Salt: 2017.7.0-1840-g1b9cb26
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Jul 12 2017, 17:32:34)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: centos 6.7 Final
         locale: UTF-8
        machine: x86_64
        release: 2.6.32-696.6.3.el6.x86_64
         system: Linux
        version: CentOS 6.7 Final
```

**Comment by rallytime** [comment]
@davidjb Ah, yes. We need to get that fixed. Thank you for letting us know. 

Just an FYI ping for @Ch3LL.

**Comment by rallytime** [comment]
Hi @davidjb - Can you give the fix in #1124 a try? This splits up the `-y` and `-x` arguments so that the Python/IUS installation won't run if `-y` isn't specified. Note however that the `-x` option assumes the proper version of Python is already installed on the host.

Since `-x` no longer depends on `-y`, the following should work in your use case without having to pass `-r`:
```
./bootstrap-salt.sh -x Python2.7 git v2017.7.0
```
And the pip dependencies will be installed with Python2.7 without hitting the IUS repo section or the Python installation again.

**Comment by davidjb** [comment]
@rallytime That helps, but requires an active decision on the part of the person (or in my case, Vagrant provisioner) whether or not to use the `-y` option.   In my case, this would require Vagrant to somehow know whether `ius-release` is already installed, which it can't know with certainty as that's within the VM.

Is it possible to have bootstrap-salt.sh test if ius-release is already installed when `-y` is used, or alternatively swallow the "error" `yum-install` produces when it states that the correct version of `ius-release` is already installed?

(FYI I needed to add `-P` to:
```
./bootstrap-salt.sh -P -x python2.7 git v2017.7.0
```
as the script errors with `ERROR: pip based installations were not allowed. Retry using '-P'` without it)
